File: ios/Plugin/HomeIndicatorPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(HomeIndicatorPlugin, "HomeIndicator",
          CAP_PLUGIN_METHOD(hide, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(show, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(isHidden, CAPPluginReturnPromise);
          CAP_PLUGIN_METHOD(getPluginVersion, CAPPluginReturnPromise);
)

File: ios/Plugin/HomeIndicatorPlugin.swift
import Foundation
import Capacitor

extension CAPBridgeViewController {

    private struct Holder {
        static var _setHomeIndicator: Bool = false
    }

    public func getHomeIndicator() -> Bool {
        return Holder._setHomeIndicator
    }

    public func hideHomeIndicator(_ ishomeIndicatorVisible: Bool) {
        Holder._setHomeIndicator = ishomeIndicatorVisible
        self.setNeedsUpdateOfHomeIndicatorAutoHidden()
    }

    override public var prefersHomeIndicatorAutoHidden: Bool {

        return Holder._setHomeIndicator

    }
}

@objc(HomeIndicatorPlugin)
public class HomeIndicatorPlugin: CAPPlugin {

    private let PLUGIN_VERSION = "0.1.8"

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve(["version": self.PLUGIN_VERSION])
    }
    @objc func hide(_ call: CAPPluginCall) {
        guard let bridgeVC = self.bridge?.viewController as? CAPBridgeViewController else {
            call.reject("")
            return
        }
        DispatchQueue.main.async {
            bridgeVC.hideHomeIndicator(true)
            call.resolve()
        }
    }

    @objc func show(_ call: CAPPluginCall) {
        guard let bridgeVC = self.bridge?.viewController as? CAPBridgeViewController else {
            call.reject("")
            return
        }
        DispatchQueue.main.async {
            bridgeVC.hideHomeIndicator(false)
            call.resolve()
        }
    }

    @objc func isHidden(_ call: CAPPluginCall) {
        guard let bridgeVC = self.bridge?.viewController as? CAPBridgeViewController else {
            call.reject("")
            return
        }
        call.resolve(["hidden": bridgeVC.getHomeIndicator()])
    }
}
